prime-router/docs/proposals/0006-email-notification/0006-email-notification.md

```diff
@@ -0,0 +1,174 @@
+# EmailService proposal
+
+## Background
+It has been recognized that notifications (in particular via email) would be a desired feature to add to ReportStream.  Toward, this end - we have secured the use of SendGrid to manage the email process and provide a well-known source of emails.  But this environment is not sufficient to address all the needs.  While SendGrid can maintain templates and send emails - it doesn't have the necessary access to ReportStream data (which might lead to duplication of data) as well, it lacks any scheduler to send messages on a periodic basis.  For these reasons - it is proposed to build an EmailService.
+
+## Goals
+
+The EmailService should:
+
+1. Respond to known and periodic events to send templated emails
+1. Maintain a base of templates that can be updated as needs change
+1. Maintain a schedule of emails to be sent
+1. Have parameters that can be filled with the template based on
+    - most recently sent emails
+    - organization and organization participants
+    - rellevant dates
+
+## Proposal
+
+The EmailService will run as a HTTP triggered service presenting a REST-ful API to manage EmailSchedules.  It will have capabilities to LIST, CREATE, DELETE, and UPDATE EmailSchedules.  When an EmailSchedule is created, the EmailService will programatically create and launch a time-trigger based Azure function based on the 'schedule' element within the supplied message for building the desired email and notifying SendGrid.  For 'send-immediate' messages, no function will be created - but the email will be sent.
+
```

Comment:
It seems like there are two types of functions: a set of HTTP API functions and a timer-triggered function.  The API function doesn't launch another function right? The API functions manipulate database records. While the timer-triggered function talks to SendGrid.